subprojects/dependency-management/src/main/java/org/gradle/internal/component/local/model/MissingLocalArtifactMetadata.java

```diff
@@ -98,4 +98,9 @@ public boolean equals(Object obj) {
         MissingLocalArtifactMetadata other = (MissingLocalArtifactMetadata) obj;
         return other.componentIdentifier.equals(componentIdentifier) && other.name.equals(name);
     }
+
+    @Override
+    public boolean isOptionalArtifact() {
+        return false;
+    }
```

Comment:
@tresat Is this still necessary?  It doesn't seem to have an effect on the integration test we added and I'm not sure I understand what caused us to add this in the first place.  Unless you know of a good reason to keep it, I'm inclined to pull it out and see if anything squawks...

Reply:
👍 I removed it.